platforms/cuda/src/CudaKernels.cpp

```diff
@@ -1796,21 +1880,44 @@ void CudaCalcNonbondedForceKernel::initialize(const System& system, const Nonbon
                 pmeFinishSpreadChargeKernel = cu.getKernel(module, "finishSpreadCharge");
                 cuFuncSetCacheConfig(pmeSpreadChargeKernel, CU_FUNC_CACHE_PREFER_L1);
                 cuFuncSetCacheConfig(pmeInterpolateForceKernel, CU_FUNC_CACHE_PREFER_L1);
+                if(doLJPME){
+                    pmeDispersionFinishSpreadChargeKernel = cu.getKernel(module, "finishSpreadC6");
+                    pmeDispersionGridIndexKernel = cu.getKernel(module, "findAtomDispersionGridIndex");
+                    pmeDispersionSpreadChargeKernel = cu.getKernel(module, "gridSpreadC6");
+                    pmeDispersionConvolutionKernel = cu.getKernel(module, "reciprocalDispersionConvolution");
+                    pmeEvalDispersionEnergyKernel = cu.getKernel(module, "gridEvaluateDispersionEnergy");
+                    pmeInterpolateDispersionForceKernel = cu.getKernel(module, "gridInterpolateDispersionForce");
+                    cuFuncSetCacheConfig(pmeDispersionSpreadChargeKernel, CU_FUNC_CACHE_PREFER_L1);
+                }
 
                 // Create required data structures.
 
                 int elementSize = (cu.getUseDoublePrecision() ? sizeof(double) : sizeof(float));
                 directPmeGrid = new CudaArray(cu, gridSizeX*gridSizeY*gridSizeZ, cu.getComputeCapability() >= 2.0 ? 2*elementSize : 2*sizeof(long long), "originalPmeGrid");
                 reciprocalPmeGrid = new CudaArray(cu, gridSizeX*gridSizeY*gridSizeZ, 2*elementSize, "reciprocalPmeGrid");
                 cu.addAutoclearBuffer(*directPmeGrid);
+                if(doLJPME){
+                    directDispersionPmeGrid = new CudaArray(cu, dispersionGridSizeX*dispersionGridSizeY*dispersionGridSizeZ,
+                                            cu.getComputeCapability() >= 2.0 ? 2*elementSize : 2*sizeof(long long), "originalDispersionPmeGrid");
+                    cu.addAutoclearBuffer(*directDispersionPmeGrid);
+                    reciprocalDispersionPmeGrid = new CudaArray(cu, dispersionGridSizeX*dispersionGridSizeY*dispersionGridSizeZ,
+                                                                2*elementSize, "reciprocalDispersionPmeGrid");
+                }
```

Comment:
We can reuse the same arrays for Coulomb and LJ PME.  Just make sure to allocate them large enough for whichever one is bigger, then clear the direct grid before starting the second one.  Likewise for pmeAtomDispersionGridIndex: we can reuse the same array.  This will save a lot of GPU memory.